### PR TITLE
installation: add option to set a custom image when creating/updating

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary
Add option to set a custom image when creating and/or updating a MM installation
For some context please see [here](https://github.com/mattermost/mattermost-server/pull/14044)

Also, this is a good addition in case we have a customer that wants to use a custom MM image

### CODE REVIEW

**Comment by cpanato** [comment]
@gabrieljackson thanks! i knew i was missing something 😓

File: cmd/cloud/installation.go [hunk 1]
```diff
@@ -13,6 +13,7 @@ func init() {
 
 	installationCreateCmd.Flags().String("owner", "", "An opaque identifier describing the owner of the installation.")
 	installationCreateCmd.Flags().String("version", "stable", "The Mattermost version to install.")
+	installationCreateCmd.Flags().String("image", "mattermost/mattermost-enterprise-edition", "The Mattermost Container iamge to use.")
```
Comment (gigawhitlocks) [inline cmd/cloud/installation.go]: s/iamge/image

**Comment by gabrieljackson** [comment]
@cpanato, I know that you haven't re-requested my review yet, but here are some follow-up thoughts that we will need to address:
1. We should add a new `image` column in the group table as well. Otherwise, it would be possible to set a version in the group that does not correlate correctly to the image value of the installation in the group. i.e. We should be able to set both image and version on both the group and the installation.
2. I see the create-installation image command flag, but the update-installation command flag still needs to be added.

**Comment by cpanato** [comment]
too many changing parts :/

**Comment by cpanato** [comment]
@gabrieljackson added in the groups, please let me know if is missing somethings else

File: cmd/cloud/group.go [hunk 2]
```diff
@@ -19,6 +19,7 @@ func init() {
 	groupUpdateCmd.Flags().String("name", "", "A unique name describing this group of installations.")
 	groupUpdateCmd.Flags().String("description", "", "An optional description for this group of installations.")
 	groupUpdateCmd.Flags().String("version", "", "The Mattermost version for installations in this group to target.")
+	groupUpdateCmd.Flags().String("image", "mattermost/mattermost-enterprise-edition", "The Mattermost Container image to use.")
```
[the remark this answers is not in the document]
Comment (cpanato) [reply cmd/cloud/group.go]: should we make those flags that are shared over several commands a root flag?

File: model/group_request.go [hunk 3]
```diff
@@ -31,6 +32,9 @@ func NewCreateGroupRequestFromReader(reader io.Reader) (*CreateGroupRequest, err
 	if createGroupRequest.Version == "" {
 		return nil, errors.New("must specify version")
 	}
+	if createGroupRequest.Image == "" {
+		return nil, errors.New("must specify image")
+	}
```
[the remark this answers is not in the document]
Comment (cpanato) [reply model/group_request.go]: it does not have similar validations

**Comment by gabrieljackson** [comment]
Sorry @cpanato, it looks like there is a merge conflict we will need to fix and then I will review asap.

**Comment by cpanato** [comment]
another one :(

File: internal/store/migrations.go [hunk 4]
```diff
@@ -728,6 +728,171 @@ var migrations = []migration{
...
+				ADD COLUMN Image TEXT NULL;
+				`)
+			if err != nil {
+				return err
+			}
+
+			_, err = e.Exec(`
+				UPDATE "Group"
+				SET Image = 'mattermost/mattermost-enterprise-edition';
+		 `)
```
[the remark this answers is not in the document]
Comment (cpanato) [reply internal/store/migrations.go]: done

**Pull request merged**